### PR TITLE
configure Client to allow_local_remote_server based on 'gss.allow_local_address'

### PR DESCRIPTION
**Pull request by daita**

### CODE REVIEW

**Comment by daita** [comment]
@MorrisJobke I was thinking of something like that

**Comment by MorrisJobke** [comment]
Makes sense 👍 

Just as background information:

For the global scale setup a local lookup server is used. If this server has a private IP address the HTTP client blocks access to it. There is a global setting that allows access to the local network, but then this would be the case for all HTTP requests from the server. As this is a security measurement it would be good to have this more fine granular. THerefore a setting that only affects the global site selector app was introduced to allow there the lookup server being in a private IP namespace and not blocked while other HTTP requests still don't allow this.

@LukasReschke I would like to get your feedback here.

**Comment by LukasReschke** [comment]
The GSS address is admin-configured and can't be changed by regular users, correct?

If so I believe we may even be fine just always defaulting to `allow_local_address` here. The issues that we are aiming to prevent here is mainly that users will be unable to issue arbitrary HTTP requests. There are some places in Nextcloud currently where this is unfortunately by design. (proxy for mail and avatar images, federated sharing etc.)

We don't want to allow regular users to be able to trigger connections to arbitrary internal hosts. But if this is an admin setting and a user can't influence the URL I'd be okay with just defaulting here to "always allow".

**Comment by LukasReschke** [comment]
In any case: The change this way or default allow would both be fine for me :)

**Comment by daita** [comment]
ok, I forced the config to true, so no need to add anything in `config.php`

File: lib/Lookup.php [hunk 1]
```diff
@@ -43,6 +43,9 @@ class Lookup {
 	/** @var  string */
 	private $lookupServerUrl;
 
+	/** @var IConfig */
+	private $config;
+
```
Comment (LukasReschke) [inline lib/Lookup.php:48]:
```suggestion
```

File: lib/Lookup.php [hunk 2]
```diff
@@ -63,6 +66,7 @@ public function __construct(IClientService $clientService,
 								ICloudIdManager $cloudIdManager
 	) {
 		$this->httpClientService = $clientService;
+		$this->config = $config;
```
Comment (LukasReschke) [inline lib/Lookup.php:69]:
```suggestion
```

**Review by LukasReschke — APPROVED** [review]

**Comment by daita** [comment]
and squashed!

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**